File: fix_corruption.py
import os
import glob

# Find all PHP files
files = glob.glob('E:/Codebytushu/**/*.php', recursive=True)
count = 0

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # Look for the exact corruption marker
    idx = content.find("<?php require_once <?php")
    
    if idx != -1:
        print(f"Fixing {filepath}...")
        
        # We need to figure out the depth to require ai-widget-loader.php properly
        # Count the slashes in the path relative to public_html root
        rel_path = os.path.relpath(filepath, 'E:/Codebytushu')
        depth = rel_path.count(os.sep)
        
        # Depth 0 (root): __DIR__ . '/includes/ai-widget-loader.php'
        # Depth 1 (Leetcode/): __DIR__ . '/../includes/ai-widget-loader.php'
        # Depth 2 (store/checkout/): __DIR__ . '/../../includes/ai-widget-loader.php'
        if depth == 0:
            require_str = "__DIR__ . '/includes/ai-widget-loader.php'"
        elif depth == 1:
            require_str = "__DIR__ . '/../includes/ai-widget-loader.php'"
        elif depth == 2:
            require_str = "__DIR__ . '/../../includes/ai-widget-loader.php'"
        else:
            prefix = '/..' * depth
            require_str = "__DIR__ . '" + prefix + "/includes/ai-widget-loader.php'"
            
        # The correct ending should just be the require statement and closing HTML
        correct_ending = f"<?php require_once {require_str}; ?>\n</body>\n</html>"
        
        # Truncate at the corruption point and append the correct ending
        new_content = content[:idx] + correct_ending
        
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        
        count += 1

print(f"Fixed {count} files successfully!")

File: fix_corruption.js
const fs = require('fs');
const path = require('path');

function findPhpFiles(dir, fileList = []) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const filePath = path.join(dir, file);
        if (fs.statSync(filePath).isDirectory()) {
            if (!filePath.includes('vendor') && !filePath.includes('node_modules')) {
                findPhpFiles(filePath, fileList);
            }
        } else if (filePath.endsWith('.php')) {
            fileList.push(filePath);
        }
    }
    return fileList;
}

const allFiles = findPhpFiles('E:/Codebytushu');
let count = 0;

for (const filePath of allFiles) {
    let content = fs.readFileSync(filePath, 'utf-8');
    const idx = content.indexOf('<?php require_once <?php');
    
    if (idx !== -1) {
        console.log(`Fixing ${filePath}...`);
        
        // Calculate depth
        const relPath = path.relative('E:/Codebytushu', filePath);
        const depth = relPath.split(path.sep).length - 1;
        
        let requireStr = '';
        if (depth === 0) {
            requireStr = "__DIR__ . '/includes/ai-widget-loader.php'";
        } else if (depth === 1) {
            requireStr = "__DIR__ . '/../includes/ai-widget-loader.php'";
        } else if (depth === 2) {
            requireStr = "__DIR__ . '/../../includes/ai-widget-loader.php'";
        } else {
            const prefix = '/..'.repeat(depth);
            requireStr = `__DIR__ . '${prefix}/includes/ai-widget-loader.php'`;
        }
        
        const correctEnding = `<?php require_once ${requireStr}; ?>\n</body>\n</html>`;
        const newContent = content.substring(0, idx) + correctEnding;
        
        fs.writeFileSync(filePath, newContent, 'utf-8');
        count++;
    }
}

console.log(`Fixed ${count} files successfully!`);
